### PR TITLE
Add Footer to the TodoMVC and Bring back Counter example

### PR DESCRIPTION
**Pull request by vramana**

### CODE REVIEW

**Comment by vramana** [comment]
@ooflorent Can you take a look at this PR ??

The current examples  fail to lint correctly due to a bug in babel-eslint I guess.  What is better way to go disable the rule for now or change the example source slightly ?

I prefer disabling the rule `block-scoped-vars` which is giving this error . Since we don't use var in  the source code, I think it is okay to turn it off.

File: examples/todomvc/components/MainSection.js [hunk 1]
```diff
@@ -1,30 +1,73 @@
...
     actions: PropTypes.object.isRequired
   };
 
+  constructor(props, context) {
+    super(props, context);
+    this.state = { showing: 'all' };
+  }
+
+  handleClearMarked() {
+    const atleastOneMarked = this.props.todos.some(todo => todo.marked);
```
Comment (ooflorent) [inline examples/todomvc/components/MainSection.js]: Can you rename the variable to `atLeastOneMarked `?

File: examples/todomvc/stores/todos.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { ADD_TODO, DELETE_TODO, EDIT_TODO, MARK_TODO, MARK_ALL } from '../constants/ActionTypes';
+import { ADD_TODO, DELETE_TODO, EDIT_TODO } from '../constants/ActionTypes';
+import { MARK_TODO, MARK_ALL, CLEAR_MARKED } from '../constants/ActionTypes';
```
Comment (ooflorent) [inline examples/todomvc/stores/todos.js]: Can you combine those 2 lines?

File: examples/todomvc/components/MainSection.js [hunk 3]
```diff
@@ -1,30 +1,73 @@
...
+    }
+
+    let footer = null;
+    if (markedCount || unmarkedCount) {
+      footer = (
+        <Footer markedCount={markedCount}
+                unmarkedCount={unmarkedCount}
+                showing={this.state.showing}
+                onClearMarked={::this.handleClearMarked}
+                onShow={(e, show) => this.handleShow(e, show)} />
```
Comment (ooflorent) [inline examples/todomvc/components/MainSection.js]: `::this.handleShow` should be enough

File: examples/todomvc/stores/todos.js (hunk 2 above)
Comment (emmenko) [inline examples/todomvc/stores/todos.js]: Or simply do 

```js
import * as constants from '../constants/ActionTypes'

constants.ADD_TODO
// ...
```

Comment (vramana) [inline examples/todomvc/stores/todos.js]: I prefer the one line but It kind of too long so I split it up into two lines.

File: examples/todomvc/components/MainSection.js (hunk 3 above)
Comment (johanneslumpe) [inline examples/todomvc/components/MainSection.js]: Actually rebinding in render is wasteful :P

Comment (gaearon) [inline examples/todomvc/components/MainSection.js]: I'd say let's not care about micro-optimizations at this point.
Sadly ES7 property syntax won't work with hot reloading (at least yet).

File: examples/todomvc/components/MainSection.js [hunk 4]
```diff
@@ -1,30 +1,73 @@
...
 
-export default class MainSection {
   static propTypes = {
     todos: PropTypes.array.isRequired,
     actions: PropTypes.object.isRequired
   };
 
+  constructor(props, context) {
+    super(props, context);
+    this.state = { showing: 'all' };
```
Comment (gaearon) [inline examples/todomvc/components/MainSection.js]: Please make these strings constants.

**Pull request merged**